src/components/link/index.tsx

```diff
@@ -0,0 +1,66 @@
+import NextLink from 'next/link'
+import { useId } from '@react-aria/utils'
+import { LinkProps } from './types'
+import { validateProps } from './helpers'
+
+const ARIA_DESCRIBED_BY_PREFIX = 'opens-in-new-tab-label'
+
+/**
+ * A wrapper for the next/link component that also handles unique functionality
+ * with the `opensInNewTab` prop.
+ *
+ * When `opensInNewTab` is passed, a `<span>` with the `g-screen-reader-only`
+ * class is rendered with "(opens in new tab)" as its inner text. This approach
+ * is inspired by WCAG 2.1 Ttechnique G201: Giving users advanced warning when
+ * opening a new window.
+ *
+ * https://www.w3.org/WAI/WCAG21/Techniques/general/G201
+ */
+const Link = ({
+	children,
+	href,
+	nextLinkProps,
+	opensInNewTab,
+	...restProps
+}: LinkProps) => {
+	validateProps({ opensInNewTab, target: restProps.target })
+
+	const uniqueId = useId()
+	const opensInNewTabLabelId = `${ARIA_DESCRIBED_BY_PREFIX}-${uniqueId}`
+	const shouldRenderScreenReaderOnlyMessage =
+		opensInNewTab === true || restProps.target === '_blank'
+
+	/**
+	 * Generate the final link's `target` prop.
+	 */
+	const target = opensInNewTab ? '_blank' : restProps.target
+
+	/**
+	 * Generate the final link's `aria-describedby` prop.
+	 */
+	let ariaDescribedBy: LinkProps['aria-describedby']
+	if (shouldRenderScreenReaderOnlyMessage) {
+		ariaDescribedBy += opensInNewTabLabelId
+	}
+	if (restProps['aria-describedby']?.length > 0) {
+		ariaDescribedBy += ` ${restProps['aria-describedby']}`
+	}
+
+	return (
+		<>
+			<NextLink href={href} {...nextLinkProps}>
+				<a {...restProps} aria-describedby={ariaDescribedBy} target={target}>
```

Comment:
[suggestion] I don't think the `nextLinkProps` separation is necessary here. For a few reasons:

1. It's not currently in use anywhere
2. With next 13, [`NextLink` renders an anchor tag automatically](https://nextjs.org/docs/api-reference/next/link), so the spread would then look like `<NextLink {...restProps} {...nextLinkProps}`
3. Based on the types, we're "extending" the native anchor element and allowing some props to be passed through to `NextLink`. This sort of makes sense when the two were separate. However, with the new behavior I think it will feel more intuitive to use if we treat is as just an extension of `NextLink`.

Extending `NextLink` should make for cleaner types as well, instead of mixing the native anchor props and the `NextLink` props. What do you think?

Reply:
Sweet! I didn't know we would be upgrading this soon, but went ahead and removed `nextLinkProps` for now in https://github.com/hashicorp/dev-portal/pull/1350/commits/454db9ed484dddf51159009e5f2cc26b30c4c8e5. 😊